libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/OfflineRoute.kt

```diff
@@ -0,0 +1,238 @@
+package com.mapbox.navigation.route.onboard
+
+import android.net.Uri
+import androidx.annotation.FloatRange
+import com.mapbox.navigation.base.route.RouteUrl
+import com.mapbox.navigation.utils.extensions.ifNonNull
+
+/**
+ * The [OfflineRoute] class wraps the [NavigationRoute] class with parameters which
+ * could be set in order for an offline navigation session to successfully begin.
+ */
+class OfflineRoute
+private constructor(
+    private val routeUrl: RouteUrl,
+    bicycleType: OfflineCriteria.BicycleType?,
+    private val cyclingSpeed: Float?,
+    private val cyclewayBias: Float?,
+    private val hillBias: Float?,
+    private val ferryBias: Float?,
+    private val roughSurfaceBias: Float?,
+    waypointTypes: List<OfflineCriteria.WaypointType?>?
+) {
+    private val bicycleType: String?
+    private val waypointTypes: String?
+
+    init {
+        this.bicycleType = bicycleType?.type
+        this.waypointTypes = checkWaypointTypes(waypointTypes)
+    }
+
+    companion object {
+
+        private const val BICYCLE_TYPE_QUERY_PARAMETER = "bicycle_type"
+        private const val CYCLING_SPEED_QUERY_PARAMETER = "cycling_speed"
+        private const val CYCLEWAY_BIAS_QUERY_PARAMETER = "cycleway_bias"
+        private const val HILL_BIAS_QUERY_PARAMETER = "hill_bias"
+        private const val FERRY_BIAS_QUERY_PARAMETER = "ferry_bias"
+        private const val ROUGH_SURFACE_BIAS_QUERY_PARAMETER = "rough_surface_bias"
+        private const val WAYPOINT_TYPES_QUERY_PARAMETER = "waypoint_types"
+
+        /**
+         * Build a new [OfflineRoute] object with the proper offline navigation parameters already setup.
+         *
+         * @return a [Builder] object for creating this object
+         */
+        @JvmStatic
+        fun builder(routeUrl: RouteUrl): Builder {
+            return Builder(routeUrl)
```

Comment:
We're having a similar problem as we had with Offboard module 👀 https://github.com/mapbox/mapbox-navigation-android/pull/2240#discussion_r350990101 https://github.com/mapbox/mapbox-navigation-android/pull/2286 The specific offline options (`bicycleType`, `cyclingSpeed`, etc.) cannot be set from a client point of view nor are included as part of `RouteOptionsNavigation`. Shouldn't we include them so they can be set (if available) in `MapboxOnboardRouter#getRoute` https://github.com/mapbox/mapbox-navigation-android/pull/2300/files#diff-1d230868e6aacaaa2ec86ec2e58a2665R63 What do you think?

Reply:
make `OfflineRoute` internal